VMEncryption/main/ResourceDiskUtil.py

```diff
@@ -0,0 +1,274 @@
+
+    def mount_key_volume(self):
+        """ attempt to mount the key volume and verify existence of key file"""
+        if not os.path.exists(self.RD_KEY_FILE):
+            self.disk_util.make_sure_path_exists(self.RD_KEY_FILE_MOUNT_POINT)
+            key_volume_device_name = os.popen('blkid -L "' + self.RD_KEY_VOLUME_LABEL + '"').read().strip()
+            self.disk_util.mount_filesystem(key_volume_device_name, self.RD_KEY_FILE_MOUNT_POINT)
+        return os.path.exists(self.RD_KEY_FILE)
+        
+    def mount(self):
+        """ mount the file system previously made on top of the crypt layer """
+        #ensure that resource disk mount point directory has been created
+        cmd = 'mkdir -p ' + self.RD_MOUNT_POINT
+        if self.executor.Execute(cmd, suppress_logging=True) != CommonVariables.process_success:
+            self.logger.log(msg='Failed to precreate mount point directory: ' + cmd, level=CommonVariables.ErrorLevel)
+            return False
+
+        # mount to mount point directory
+        mount_result = self.disk_util.mount_filesystem(dev_path=self.mapper_path, mount_point=self.RD_MOUNT_POINT, file_system=CommonVariables.default_file_system)
+        if mount_result != CommonVariables.process_success:
+            self.logger.log(msg="Failed to mount file system on resource disk", level=CommonVariables.ErrorLevel)
+            return False
+        return True
+
+    def configure_waagent(self):
+        """ turn off waagent.conf resource disk management  """ 
+        # set ResourceDisk.MountPoint to standard /mnt mount point
+        cmd = "sed -i.bak 's|ResourceDisk.MountPoint=.*|ResourceDisk.MountPoint=/mnt|' /etc/waagent.conf"
+
+        # set ResourceDiskFormat=n to ensure waagent does not attempt a simultaneous format
+        cmd = "sed -i.bak 's|ResourceDisk.Format=y|ResourceDisk.Format=n|' /etc/waagent.conf"
+        if self.executor.ExecuteInBash(cmd) != CommonVariables.process_success:
+            self.logger.log(msg="Failed to set ResourceDiskFormat in /etc/waagent.conf", level=CommonVariables.WarningLevel)
+            return False
+        # todo: restart waagent if necessary to ensure changes are picked up?
+        return True
+
+    def configure_fstab(self):
+        """ remove resource disk from /etc/fstab if present """
+        cmd = "sed -i.bak '/azure_resource-part1/d' /etc/fstab"        
+        if self.executor.ExecuteInBash(cmd) != CommonVariables.process_success:
+            self.logger.log(msg="Failed to configure resource disk entry of /etc/fstab", level=CommonVariables.WarningLevel)
+            return False        
+        return True
+
+    def unmount_resource_disk(self):
+        """ return true if mount point is mounted regardless of crypt status """
+        # after service healing multiple unmounts of key file mount point may be required
+        self.disk_util.umount(self.RD_KEY_FILE_MOUNT_POINT)
+        self.disk_util.umount(self.RD_KEY_FILE_MOUNT_POINT)
```

Comment:
Not sure why the line is repeated.. Has it got to do with the comment right above?

Reply:
When VM is restored from a service healing scenario, although I do not yet understand why, a single unmount of /mnt/azure_bek_disk is not sufficient, however on a second call it is unmounted, which is required prior to the unmount of /mnt to succeed.